kubernetes/base/stream/ws_client.py

```diff
@@ -179,7 +179,9 @@ def update(self, timeout=0):
         #                   efficient as epoll. Will work for fd numbers above 1024.
         # select.epoll()  - newest and most efficient way of polling.
         #                   However, only works on linux.
-        if sys.platform.startswith('linux') or sys.platform in ['darwin']:
+        #                   Will not work on Linux when eventlet is being used.
+        if ((sys.platform.startswith('linux') and getattr(select, "poll", None) is not None) or
```

Comment:
eventlet supports epoll, does it make sense to use epoll

Reply:
If that's a better solution I'm happy to rework the PR.  I'm not very familiar with select and poll but if epoll is equivalent I'll update the change.